lxd/main_init_interactive.go

```diff
@@ -116,8 +116,17 @@ func (c *cmdInit) askClustering(config *cmdInitData, d lxd.InstanceServer) error
 
 		// Cluster server address
 		address := util.NetworkInterfaceAddress()
-		serverAddress := util.CanonicalNetworkAddress(cli.AskString(
-			fmt.Sprintf("What IP address or DNS name should be used to reach this node? [default=%s]: ", address), address, nil))
+		var serverAddress string
+		for {
+			serverAddress = util.CanonicalNetworkAddress(cli.AskString(
+				fmt.Sprintf("What IP address or DNS name should be used to reach this node? [default=%s]: ", address), address, nil))
+			host, _, _ := net.SplitHostPort(serverAddress)
+			if shared.StringInSlice(host, []string{"", "[::]", "0.0.0.0"}) {
+				fmt.Printf("Invalid IP address or DNS name\n")
+				continue
+			}
+			break
+		}
```

Comment:
Wouldn't this be simpler as a validator function passed as the last argument to `AskString`?

Reply:
Good call. Fixed.